core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -12,50 +12,31 @@
  */
 package tech.pegasys.web3signer.core.service.jsonrpc.handlers.signing;
 
-import static tech.pegasys.web3signer.core.service.jsonrpc.response.JsonRpcError.SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT;
-
-import tech.pegasys.web3signer.core.service.jsonrpc.exceptions.JsonRpcException;
 import tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.transaction.Transaction;
-import tech.pegasys.web3signer.signing.ArtifactSignerProvider;
+import tech.pegasys.web3signer.signing.ArtifactSigner;
 import tech.pegasys.web3signer.signing.SecpArtifactSignature;
 import tech.pegasys.web3signer.signing.secp256k1.Signature;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
 import org.web3j.crypto.Sign.SignatureData;
 import org.web3j.crypto.TransactionEncoder;
 import org.web3j.utils.Numeric;
 
 public class TransactionSerializer {
-  private static final Logger LOG = LogManager.getLogger();
 
-  protected final ArtifactSignerProvider signer;
+  protected final ArtifactSigner signer;
   protected final long chainId;
 
-  protected final String identifier;
-
-  public TransactionSerializer(
-      final ArtifactSignerProvider signer, final long chainId, final String identifier) {
+  public TransactionSerializer(final ArtifactSigner signer, final long chainId) {
     this.signer = signer;
     this.chainId = chainId;
-    this.identifier = identifier;
   }
 
   public String serialize(final Transaction transaction) {
     final byte[] bytesToSign = transaction.rlpEncode(chainId);
 
     final SecpArtifactSignature artifactSignature =
-        signer
-            .getSigner(identifier)
-            .map(
-                artifactSigner ->
-                    (SecpArtifactSignature) artifactSigner.sign(Bytes.of(bytesToSign)))
-            .orElseThrow(
-                () -> {
-                  LOG.debug("From address ({}) does not match any available account", identifier);
-                  throw new JsonRpcException(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT);
-                });
+        (SecpArtifactSignature) signer.sign(Bytes.of(bytesToSign));
```

Comment:
Doing this would mean creating a new SignerForIdentifier with the signerProviderAdapted just to avoid the cast which is done the same way but in a more elegant way using the generics in the SignerForIdentifier class in the formatSignature method. I see this cast as a safe cast but happy to switch if we're strongly against the cast.